### ISSUE TITLE
Support CSS selector by element's inner text/content

### ISSUE THREAD

**Issue by noperator** [mgdm/htmlq#55]
Love `htmlq`—use it daily. Thanks for the effort you've put into this tool.

I have a pretty frequent need to find an element based on the text that it contains. For example, I'd like to match on the first `div` (i.e., the one containing `Item 1`).

```html
<div>Item 1</div>
<div>Item 2</div>
<div>Item 3</div>
```

There have been various attempts at updating the official CSS specification to support this kind of functionality, though I don't think any of them have actually made their way into the spec. Instead, various tools (Playwright. etc.) extend their own CSS selectors to support one of the following forms.

```css
div:contains("Item 1")
div:has-text("Item 1")
div[innerText="Item 1"]
div[textContent="Item 1"]
```

It would be very useful if I could use the following CSS selector with `htmlq` to match on the previously noted element.

```bash
htmlq 'div:contains("Item 1")'
```

To my understanding, `htmlq` relies on [kuchiki](https://github.com/kuchiki-rs/kuchiki) for HTML parsing, which in turn relies on [servo](https://github.com/servo/servo/tree/master/components/selectors) for CSS selection—so I think I'd need to request support for this upstream in `servo`. Does that seem right to you? Just wanted to run this idea past you in case you've thought about it already and/or have an opinion about it.

**Comment by harri-halttunen-aktia** [comment]
I highly support the idea of selecting elements based on their content. While it is true that `:contains()` [did not make it way to CSS3](https://www.w3.org/TR/selectors-3/#content-selectors) (as it will break the whole idea of separating structure and content) it would be extremely important feature for tools like `htmlq`or `pup` as the content to be filtered cannot be controlled by the user of these tools.

Acutally, `pup` has this feature but it unfortunately does not have a general sibling selector which `htmlq` has. It would be nice to have both features.

**Comment by noperator** [comment]
Agree. I'm kind of doubtful that a widely used package like `servo` would accept a PR for a nonstandard CSS selector like `:contains()`. More realistic option might be to find a way to implement it downstream in `kuchiki`, or directly within `htmlq`.

**Comment by noperator** [comment]
[Drafted a change](https://github.com/mgdm/htmlq/compare/master...noperator:htmlq:contains-text#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fc) as proposed above. Given the following HTML sample from https://lethain.com/company-team-self/:

```html
<li class="mb2">
  <a href="/work-hard-work-smart/">
    Work hard / work smart.</a>
</li>
<li class="mb2">
  <a href="/mailbag-not-measurable-whether-hire-exec/">  👈
    Mailbag: What isn't measurable? To hire as exec or not?</a>
</li>
<li class="mb2">
  <a href="/reminiscing/">
    Reminiscing: the retreat to comforting work.</a>
</li>
```

You can find the hyperlink list item `li.mb2 > a` whose name matches the case-insensitive regex `(?i)mailbag.*measure?`, extract the `href`, and prepend `https://example.com` to its base URL.

```bash
curl -s https://lethain.com/company-team-self/ |
    htmlq -c '(?i)mailbag.*measure?' -a href -b https://example.com 'li.mb2 > a'

https://example.com/mailbag-not-measurable-whether-hire-exec/
```
